tsdb/engine/tsm1/reader.go

```diff
@@ -391,6 +405,10 @@ func (d *indirectIndex) Add(key string, blockType byte, minTime, maxTime time.Ti
 	panic("unsupported operation")
 }
 
+func (d *indirectIndex) Write(w io.Writer) error {
+	panic("unsupported operation")
```

Comment:
any reason to panic here instead of return an error? Guess it's just a code path that never gets hit and you want to be sure?